runtime/vm/createramclass.cpp

```diff
@@ -1759,6 +1760,7 @@ loadFlattenableFieldValueClasses(J9VMThread *currentThread, J9ClassLoader *class
 					flattenableInstanceFieldCount += 1;
 				}
 				*valueTypeFlags |= (valueClass->classFlags & (J9ClassLargestAlignmentConstraintDouble | J9ClassLargestAlignmentConstraintReference));
+				eligibleForFastSubstitutability = false;
```

Comment:
if the class is flattened and has the `J9ClassCanSupportFastSubstitutability` bit set then it is eligeable